// Code generated by mockery v1.0.0
package mocks

import mock "github.com/stretchr/testify/mock"
import reader "github.com/opsidian/parsley/reader"

// Error is an autogenerated mock type for the Error type
type Error struct {
	mock.Mock
}

// Cause provides a mock function with given fields:
func (_m *Error) Cause() error {
	ret := _m.Called()

	var r0 error
	if rf, ok := ret.Get(0).(func() error); ok {
		r0 = rf()
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// Error provides a mock function with given fields:
func (_m *Error) Error() string {
	ret := _m.Called()

	var r0 string
	if rf, ok := ret.Get(0).(func() string); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(string)
	}

	return r0
}

// Msg provides a mock function with given fields:
func (_m *Error) Msg() string {
	ret := _m.Called()

	var r0 string
	if rf, ok := ret.Get(0).(func() string); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(string)
	}

	return r0
}

// Pos provides a mock function with given fields:
func (_m *Error) Pos() reader.Position {
	ret := _m.Called()

	var r0 reader.Position
	if rf, ok := ret.Get(0).(func() reader.Position); ok {
		r0 = rf()
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(reader.Position)
		}
	}

	return r0
}
